meson.build

```diff
@@ -477,26 +480,17 @@ if get_option('mode') == 'developer'
         possible_cc_flags += '-fno-omit-frame-pointer'
 endif
 
-add_project_arguments(cc.get_supported_arguments(basic_disabled_warnings), language : 'c')
-add_project_arguments(cc.get_supported_arguments(possible_cc_flags), language : 'c')
-add_project_link_arguments(cc.get_supported_link_arguments(possible_link_flags), language : 'c')
+add_project_arguments(cc.get_supported_arguments(
+        basic_disabled_warnings,
+        possible_common_cc_flags
+), language : 'c')
+
+userspace_c_args += cc.get_supported_arguments(possible_cc_flags)
+userspace_c_ld_args += cc.get_supported_link_arguments(possible_link_flags)
 
 have = cc.has_argument('-Wzero-length-bounds')
 conf.set10('HAVE_ZERO_LENGTH_BOUNDS', have)
 
-if cc.compiles('''
-   #include <time.h>
-   #include <inttypes.h>
-   typedef uint64_t usec_t;
-   usec_t now(clockid_t clock);
-   int main(void) {
-           struct timespec now;
-           return 0;
-   }
-''', args: '-Werror=shadow', name : '-Werror=shadow with local shadowing')
-        add_project_arguments('-Werror=shadow', language : 'c')
-endif
```

Comment:
I put it in `possible_common_cc_flags` as I think it does the same?
```
        '-Wshadow', # clang
        '-Werror=shadow',
```
I think this is a relic from autotools → meson conversion, but i could be wrong.

Reply:
No, it was a custom test for that option. It wasn't done via `possible_common_cc_flags` because some compiler versions do not implement it correctly.